spark-common/src/main/java/com/k15t/spark/atlassian/AtlassianRequestProperties.java

```diff
@@ -25,8 +30,14 @@ protected String getUrlLocalPart() {
         if (urlLocalPart == null) {
             urlLocalPart = StringUtils.removeStart(request.getPathInfo(), appPrefix);
         }
-
         return StringUtils.defaultString(urlLocalPart);
     }
 
+
+    @Override
+    public URI getUri() {
+        // ... enforce to use the configured base url of the underlying system e.g. Confluence instead of url of the http request.
```

Comment:
The comment is not correct:
The requestProperties instance is created with a baseUrl of UrlMode.RELATIVE in AtlassianAppServlet. Since it is extremely probable that a request object is always available when this code path is executed, the baseUrl will be taken from there.

Either change the comment to reflect this, or pass in a baseUrl retrieved with UrlMode.RELATIVE_CANONICAL.

Maybe the comment should be moved to the class constructor's javadoc so clients may consider it. At this place the comment has no real influence on what clients pass to this class.